activerecord/lib/active_record/persistence.rb

```diff
@@ -143,7 +143,11 @@ def insert(attributes, returning: nil, unique_by: nil, record_timestamps: nil)
       #     { id: 2, title: "Eloquent Ruby" }
       #   ])
       def insert_all(attributes, returning: nil, unique_by: nil, record_timestamps: nil)
-        InsertAll.new(self, attributes, on_duplicate: :skip, returning: returning, unique_by: unique_by, record_timestamps: record_timestamps).execute
+        if attributes.empty?
```

Comment:
To prevent this wrap 3 times it would be possible as well to create something like `InsertAll.execute` and do this simple check in there.